src/foolscap/connections/tor.py

```diff
@@ -0,0 +1,167 @@
+import os, re
+from twisted.internet.interfaces import IStreamClientEndpoint
+from twisted.internet.defer import inlineCallbacks, returnValue, succeed
+import ipaddress
+from .. import observer
+
+from zope.interface import implementer
+from ..ipb import IConnectionHintHandler, InvalidHintError
+from ..util import allocate_tcp_port
+import txtorcon
```

Comment:
There's a `txtorcon.util.available_tcp_port` utility method, if this is the only place you use it and are depending on txtorcon anyway...